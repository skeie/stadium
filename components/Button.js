import React from 'react';
import {
  TouchableOpacity,
  ScrollView,
  View,
  Image,
  Dimensions,
  ActivityIndicator,
} from 'react-native';
import { Poppins } from './StyledText';

type Props = {
  style: any,
  onPress: () => void,
  children: string,
  loading?: boolean,
};

const Button = ({ style, onPress, children, loading = false }) => {
  return (
    <TouchableOpacity
      onPress={loading ? () => {} : onPress}
      style={{
        borderColor: '#553555',
        borderWidth: 1,
        alignItems: 'center',
        padding: 10,
        borderRadius: 10,
        marginBottom: 10,
        ...style,
      }}
    >
      {loading ? <ActivityIndicator color="#553555" /> : <Poppins>{children}</Poppins>}
    </TouchableOpacity>
  );
};

export default Button;
