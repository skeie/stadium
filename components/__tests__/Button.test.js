import 'react-native';
import React from 'react';
import Button from '../Button';
import renderer from 'react-test-renderer';

it('Button renders correctly', () => {
    const tree = renderer.create(<Button>Snapshot test!</Button>).toJSON();
    expect(tree).toMatchSnapshot();
});

