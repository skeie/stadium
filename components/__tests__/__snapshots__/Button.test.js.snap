// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders correctly 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  collapsable={undefined}
  hasTVPreferredFocus={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "borderColor": "#553555",
      "borderRadius": 10,
      "borderWidth": 1,
      "marginBottom": 10,
      "opacity": 1,
      "padding": 10,
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": "#553555",
          "fontFamily": "poppins",
        },
        undefined,
        undefined,
      ]
    }
  >
    Snapshot test!
  </Text>
</View>
`;
